### PR TITLE
French Localizations Update & Fixed Sileo Theme for French language

### PR DESCRIPTION
**Pull request by CorentiOS**
## Information

Hi, I updated French localizations for the project, and also fixed an issue with the theme (the strings variables were missing) and the app went on white mode on French langage.

I'm French, so it will be a pleasure to help you in the future and not only on translate ! :) 

## Type of change

- [x] Localizations updated.
- [x] Replaced some hard variables in the project with localization variable.
- [x] Fixed dark mode problem when the app was in French.

### CODE REVIEW

**Comment by CharlieWhile13** [comment]
Could I ask you to change the branch you're merging into to amy

Mentioning @Cimmerian-Iter to confirm the localisation changes

File: Sileo/Sileo/fr.lproj/Categories.strings [hunk 1]
```diff
@@ -9,7 +9,7 @@
 */
 
 "No_Category" = "(Aucune Catégorie)";
-"Tweaks" = "Modifications";
+"Tweaks" = "Tweaks";
```
Comment (Cimmerian-Iter) [inline Sileo/Sileo/fr.lproj/Categories.strings:12]: I understand why you want to do that change, but it's better to keep it in french.
1 other traduction also translated that one
2 better translate every words in french

File: Sileo/Sileo/fr.lproj/Errors.strings [hunk 2]
```diff
@@ -4,58 +4,58 @@
...
-  Updated by : Cimmerian-Iter and Dadoum
+  Updated by : Cimmerian-Iter and Dadoum and CorentiOS
   Copyright © 2018 CoolStar. All rights reserved.
  
- BASE (English) LOCALISATIONS
+ BASE (Français) LOCALISATIONS
  */
 
-"Unknown" = "Il y a eu un erreur inconnu.";
+"Unknown" = "Une erreur inconnue s'est produite.";
```
Comment (Cimmerian-Iter) [inline Sileo/Sileo/fr.lproj/Errors.strings:13]: Yeah it's better

File: Sileo/Sileo/fr.lproj/Errors.strings [hunk 3]
```diff
@@ -4,58 +4,58 @@
...
+"Form_Error.Title" = "Impossible de charger le formulaire";
 "Form_Load_Error" = "Le formulaire n'a pas pu être chargé depuis le serveur.";
-"Invalid_Form_Data" = "Le serveur a rendu des données invalides.";
-"Form_Submit_Failure" = "Un erreur s'est produit en essayant de soumettre le formulaire: %@";
-"Form_Submit_Error.Title" = "Incapable de soumettre le formulaire";
-"Form_Action_Host_Mismatch" = "L'URL hôte de l'action du formulaire ne correspond pas à celui de ce même formulaire.";
+"Invalid_Form_Data" = "Le serveur a renvoyé des données invalides.";
+"Form_Submit_Failure" = "Une erreur s'est produite lors de l'envoi du formulaire: %@";
+"Form_Submit_Error.Title" = "Impossible de soumettre le formulaire";
+"Form_Action_Host_Mismatch" = "L'URL hôte de l'action du formulaire ne correspond pas à celle de ce même formulaire.";
```
Comment (Cimmerian-Iter) [inline Sileo/Sileo/fr.lproj/Errors.strings:51]: Non URL est masculin donc on devrait garder "celui"

File: Sileo/Sileo/fr.lproj/Localizable.strings [hunk 4]
```diff
@@ -68,13 +72,13 @@
...
 
-"Package_Unavailable" = "Paquet non disponible";
+"Package_Unavailable" = "Paquet indisponible";
 "Package_By_Author" = "%@ par %@"; // first %@ = package name, second %@ = author name
 "By_Author" = "par %@"; // %@ = author name
 "Packages_By_Author" = "Paquets par %@"; // %@ = author name
 
-"Featured_Page" = "En Vedette";
-"News_Page" = "Nouveau";
+"Featured_Page" = "À l'affiche";
```
Comment (Cimmerian-Iter) [inline Sileo/Sileo/fr.lproj/Localizable.strings:80]: No, "En Vedette" is better and also used in other package managers

File: Sileo/UI/PackageViewController/PackageViewController.swift [hunk 5]
```diff
@@ -683,7 +683,7 @@ class PackageViewController: SileoViewController, PackageQueueButtonDataProvider
         let isPurchased = existingPurchased?.contains(package.package) ?? false
 
         if isPurchased {
-            let info = PaymentPackageInfo(dictionary: ["price": "Paid",
+            let info = PaymentPackageInfo(dictionary: ["price": String(localizationKey: "Package_Paid"),
```
Comment (Cimmerian-Iter) [inline Sileo/UI/PackageViewController/PackageViewController.swift:686]: Does that fix string not showing?

**Review by Cimmerian-Iter — CHANGES_REQUESTED** [review]
Other than what I mentioned the other changes are good

File: Sileo/Sileo/fr.lproj/Errors.strings (hunk 3 above)
Comment (CorentiOS) [reply Sileo/Sileo/fr.lproj/Errors.strings:51]: On dit plutôt une adresse URL : https://www.larousse.fr/dictionnaires/francais/URL/80723

File: Sileo/UI/PackageViewController/PackageViewController.swift (hunk 5 above)
Comment (CorentiOS) [reply Sileo/UI/PackageViewController/PackageViewController.swift:686]: When you are in French you keep seeing "Paid" so now you will see the correct translation

File: Sileo/Sileo/fr.lproj/Errors.strings (hunk 3 above)
Comment (Cimmerian-Iter) [reply Sileo/Sileo/fr.lproj/Errors.strings:51]: tu as raison dans ce cas

File: Sileo/UI/PackageViewController/PackageViewController.swift (hunk 5 above)
Comment (CharlieWhile13) [reply Sileo/UI/PackageViewController/PackageViewController.swift:686]: > 
> 
> When you are in French you keep seeing "Paid" so now you will see the correct translation

This is only in the simulator, but is something that should be changed

**Comment by CorentiOS** [comment]
> Could I ask you to change the branch you're merging into to amy
> 
> Mentioning @Cimmerian-Iter to confirm the localisation changes

Done

**Pull request merged**